dojo/product_type/views.py

```diff
@@ -22,70 +24,99 @@
-    return render(request, 'dojo/product_type.html', {
-        'name': 'Product Type List',
-        'metric': False,
-        'user': request.user,
-        'pts': pts,
-        'ptl': ptl,
-        'name_words': name_words})
+    return render(
+        request,
+        "dojo/product_type.html",
+        {
+            "name": "Product Type List",
+            "metric": False,
+            "user": request.user,
+            "pts": pts,
+            "ptl": ptl,
+            "name_words": name_words,
+        },
+    )
 
 
 @user_passes_test(lambda u: u.is_staff)
 def add_product_type(request):
     form = Product_TypeForm()
-    if request.method == 'POST':
+    if request.method == "POST":
         form = Product_TypeForm(request.POST)
         if form.is_valid():
             form.save()
-            messages.add_message(request,
-                                 messages.SUCCESS,
-                                 'Product type added successfully.',
-                                 extra_tags='alert-success')
-            return HttpResponseRedirect(reverse('product_type'))
+            messages.add_message(
+                request,
+                messages.SUCCESS,
+                "Product type added successfully.",
+                extra_tags="alert-success",
+            )
+            return HttpResponseRedirect(reverse("product_type"))
     add_breadcrumb(title="Add Product Type", top_level=False, request=request)
-    return render(request, 'dojo/new_product_type.html', {
-        'name': 'Add Product Type',
-        'metric': False,
-        'user': request.user,
-        'form': form,
-    })
+    return render(
+        request,
+        "dojo/new_product_type.html",
+        {
+            "name": "Add Product Type",
+            "metric": False,
+            "user": request.user,
+            "form": form,
+        },
+    )
 
 
 @user_passes_test(lambda u: u.is_staff)
 def edit_product_type(request, ptid):
     pt = get_object_or_404(Product_Type, pk=ptid)
-    form = Product_TypeForm(instance=pt)
-    if request.method == 'POST':
-        form = Product_TypeForm(request.POST, instance=pt)
-        if form.is_valid():
-            pt = form.save()
-            messages.add_message(request,
-                                 messages.SUCCESS,
-                                 'Product type updated successfully.',
-                                 extra_tags='alert-success')
-            return HttpResponseRedirect(reverse('product_type'))
+    form1 = Product_TypeForm(instance=pt)
+    form2 = Delete_Product_TypeForm(instance=pt)
+    if request.method == "POST" and request.POST.get("edit_product_type"):
+        form1 = Product_TypeForm(request.POST, instance=pt)
+        if form1.is_valid():
+            pt = form1.save()
+            messages.add_message(
+                request,
+                messages.SUCCESS,
+                "Product type updated successfully.",
```

Comment:
As recommended by [PEP- 498](https://www.python.org/dev/peps/pep-0498/), let's keep single quotes, unless double quotes are required.
```suggestion
                'Product type updated successfully.',
```
And following lines.